lib/component.js

```diff
@@ -176,16 +196,25 @@ class Component extends WebComponent {
     if (!this.initialized) {
       Object.assign(this.state, stateUpdate);
     } else if (this.isPanelRoot) {
-      const updateHash = '$fragment' in stateUpdate && stateUpdate.$fragment !== this.state.$fragment;
+      this._rootUpdate(stateUpdate);
+    } else {
+      this._rootUpdate(stateUpdate);
+      this.$panelRoot._rootUpdate(this.sharedStateEntries(stateUpdate), {exclude: this});
+    }
+  }
 
-      Object.assign(this.state, stateUpdate);
-      this.updateSelfAndChildren(this.state);
+  // Apply the given update down the component hierarchy from this node,
+  // optionally excluding one node's subtree. This is useful for applying
+  // a full state update to one component while sending only "shared" state
+  // updates to the app root.
```

Comment:
Restating: a child component can share state with its parent and still have its own private state. Did I get that right?

Reply:
Yup, that's a great way to distill it down to the essence. Components have always been able to have non-state properties that they don't share with parent (i.e. plain old JS object properties), but this PR makes it so that `this.update()` and `this.state` work seamlessly with shared vs private state.